### PR TITLE
Add documentation on requestId

### LINKED ISSUE #413: Example / Documentation for Cancel Request
The engine allows to cancel / abort a request.
Please give a hint / example how this works.

### PR DESCRIPTION
**Pull request by peol**
Fixes #413

### CODE REVIEW

**Comment by hrigner** [comment]
Should this be included in the api spec?

**Comment by peol** [comment]
@hrigner @wennmo Please re-review. I've added JSDoc for it but it's quite tricky since it's basically only accessible through a generated API which wasn't previously documented either. I've added @miralemd for sanity check as well.

File: src/schema.js [hunk 1]
```diff
@@ -174,6 +173,24 @@ class Schema {
...
 
+  /**
+   * Promise returned by invoking generated API methods.
+   * @interface GeneratedPromise
+   * @type Promise<Object>
+   * @property {Number} requestId The JSON-RPC request id.
+   */
+
+  /**
+    * API method for a specific schema method. Note that this is not
```
Comment (miralemd) [inline src/schema.js]: not really the best way to document this - the entire generated api is an object of type `object<string,generatedMethod>` with some additional custom properties that override the default property type of the object. The jsdoc for that might look something like this:

```js
/**
 * @interface api
 * @type {object<string,generatedMethod>}
 * @property {string} id
 * ...
 */
```

Unfortunately this is not something jsdoc nor scriptappy supports out of the box

Comment (peol) [reply src/schema.js]: Gotcha, tried that as well. Since that doesn't work, the way it's done in this PR _is_ the best _workable_ way, or?

Comment (miralemd) [reply src/schema.js]: I think the best workable solution right now is to not include it in the spec at all, rather than including a wrong spec.

Comment (peol) [reply src/schema.js]: Agreed, if that's acceptable. @wennmo @hrigner ?

**Pull request merged**